### PR TITLE
Merging the 6.12.0 release into master

### PR DESCRIPTION
**Pull request by maksymmalyhin**

### CODE REVIEW

File: scripts/check_whitespace.sh [hunk 1]
```diff
@@ -29,14 +29,16 @@ git grep "${options[@]}" -- \
     ':(exclude)Firebase/CoreDiagnostics/FIRCDLibrary/Protogen/nanopb' \
     ':(exclude)Firebase/CoreDiagnostics/ProtoSupport' \
     ':(exclude)CoreOnly/NOTICES' \
+    ':(exclude)Firebase/Firebase/NOTICES' \
```
Comment (paulb777) [inline scripts/check_whitespace.sh]: Still catching up, but we really should avoid a third NOTICES file.

**Comment by paulb777** [comment]
When there are non-trivial cherry-picks, I've found that it's cleaner to make a branch from master and manually merge (and resolve any conflicts) from the release branch before sending the merge-back PR.  

Otherwise github will add a bunch of changes already in master to the PR.

File: scripts/check_whitespace.sh (hunk 1 above)
Comment (maksymmalyhin) [reply scripts/check_whitespace.sh]: removed

**Comment by maksymmalyhin** [comment]
The only Travis failure is now is Messaging which is still failing in master

**Pull request merged**